### PR TITLE
Integrate React Frontend with Spring Boot Backend using Vite

### PR DESCRIPTION
**Pull request by JrSchoolAccount**
## Description
This pull request integrates the React frontend with the Spring Boot backend using Vite as the bundler for the React app.

## Changes
- Configured Vite to output build files to the backend's `static/` directory.
- Updated Spring Boot configuration to serve React's static files.
- Updated `README.md` with build instructions.

## Tasks Completed
- [x] Set up React project with Vite.
- [x] Added react-dom for frontend routing
- [x] Added Vite dependencies: Tailwindcss, postcss & autoprefixer
- [x] Updated Spring Boot configuration, WebMvcConfigurer
- [x] Building the Frontend as part of Maven lifecycle
- [x] Installing node/npm locally as part of Maven lifecycle
- [x] Verified the frontend is served correctly by the backend
- [x] Cross-browser testing.
- [x] Relative path, `resolve(__dirname, 'src/main/resources/static/')`
- [x] Update `README.md` compile instructions. 
## Notes
- Refer to [this article](https://www.jessym.com/articles/bundling-react-vite-with-spring-boot) for implementation details.
- Suggestions for improvements are welcome!

## Release Notes

- **New Features**
	- Introduced routing capabilities with navigation to a new `/cake` route.
	- Added a main entry point HTML file for the web application.
	- Enhanced styling with new CSS for improved visual consistency and accessibility.

- **Dependencies**
	- Integrated `react-router-dom` for routing.
	- Added several development tools including `tailwindcss` and `postcss`.
	- Included a plugin for managing frontend dependencies in Maven.

- **Documentation**
	- Updated README with instructions for building the frontend as part of the Maven lifecycle.

- **Configuration**
	- Updated build configurations to specify output directories and manage frontend dependencies.
	- Enhanced static resource handling in the application.

### CODE REVIEW

File: frontend/vite.config.ts
Comment (MatsJoh50) [inline frontend/vite.config.ts:1]: we might want to change the path to a more relative to always target root => src...
that way we can move or accidentely move the frontend structure and always get correct buildpath.

Ripped code from GPT:

import { resolve } from 'path';

export default {
  build: {
    outDir: resolve(__dirname, 'src/main/resources/static/'),
    emptyOutDir: true,
  },
};

**Review by MatsJoh50 — DISMISSED** [review]
Great job.

one minor suggestion that might be more trouble then reward.

works great and builds the project at correct place as it is!

File: frontend/package.json
[the remark this answers is not in the document]
Comment (JrSchoolAccount) [reply frontend/package.json:1]: It seemed odd not to include Tailwind when the frontend uses Vite.
Figured it was some sort of mishap so i added it 😊

File: frontend/vite.config.ts
Comment (JrSchoolAccount) [reply frontend/vite.config.ts:1]: Added this to pending tasks

**Review by MatsJoh50 — APPROVED** [review]
I see you implemented the "go from root" change.
_

**Pull request merged**